forge-game/src/main/java/forge/game/trigger/TriggerHandler.java

```diff
@@ -541,8 +541,9 @@ private void runSingleTriggerInternal(final Trigger regtrig, final Map<AbilityKe
                 sa.changeText();
             }
         } else {
+            Player controller = delayedTriggers.contains(regtrig) ? sa.getActivatingPlayer() : host.getController();
```

Comment:
i don't like the `delayedTriggers.contains(regtrig)` check

could we maybe look for SpawningAbility ?